### PR TITLE
fix(sdui-runtime): read flat label, resolve bff_call via EndpointPaths, lock sdk peer-dep

### PR DESCRIPTION
**Pull request by mridulgupta-oi**
## Summary

Aligns `@one-impression/sdui-runtime` with the `@one-impression/sdk-native-sdui` wire contract (the source of truth), fixing two on-device home-page crashes and closing the version gap that let the drift go uncaught.

- **FIX 1 — flat label reads (crash):** `label`/`subtitle` are declared as flat `TextSchema` (`{ text, ... }`), but 7 ui_component renderers read them as nested nodes (`v.label.data.text`), throwing "Cannot read property 'text' of undefined". Now read `v.label.text` / `v.subtitle.text`. Files: `Tab`, `Tag`, `Chip`, `Checkbox`, `Radio`, `SelectableItem`, `Input` renderers (and `SelectableItem` subtitle).
- **FIX 2 — bff_call endpoint resolution (404):** the logical endpoint id (e.g. `creator.home.tab`) was treated as a literal path → `/creator.home.tab` → 404. Now resolved through `EndpointPaths` (id → `/v1/...`), with path-param substitution on the resolved path, a trailing-slash trim on the base URL to avoid a double slash, and a loud throw if an id is unregistered.
- **FIX 3 — peer-dep lock:** `@one-impression/sdk-native-sdui` peerDependency raised `>=1.0.0` → `^2.6.0` (the version that ships `EndpointPaths`); added as a devDependency at `^2.6.0` so the import resolves at build/test time. This surfaces schema/renderer drift at install instead of on the simulator.
- **FIX 4 — anti-regression:** new emit→render contract test round-trips the home node set against their `sdk-native-sdui` schemas and asserts `tab`/`chip`/`tag`/`info_breakdown_row` built nodes expose a flat `data.label.text` (NOT `data.label.data`). Plus 4 new `bff_call` URL-resolution tests.

Changeset: `minor` for `@one-impression/sdui-runtime`.

## Test plan

- [x] `npm run build` (tsup + tsc) — 0 errors
- [x] `npm test` — 137/138 pass; the 1 failure (`node-registry.test.ts`) is a pre-existing RN-transform issue in the test harness, unrelated to this change (confirmed failing on clean `origin/main`)
- [x] New `emit-render-contract` test (4 cases) green
- [x] New `bff_call` URL-resolution tests (4 cases) green
- [ ] On-device home page renders + tab/chip taps hit the correct BFF paths

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/sdui-runtime/src/__tests__/emit-render-contract.test.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
+/**
+ * Emit→render contract test (CR-WF3).
```
Comment (zenith-oi) [inline packages/sdui-runtime/src/__tests__/emit-render-contract.test.ts]: **FLAG** — [no-task-refs-in-durable-artifacts] The file-level JSDoc comment opens with `* Emit→render contract test (CR-WF3).` — `CR-WF3` is a task/work-item reference embedded in a durable source file. Per the `no-task-refs-in-durable-artifacts` pattern, brief/task/work-item IDs belong in commit messages and PR descriptions, not in code comments that will outlive the task. Replace with a timeless description of what the test guards (e.g. 'Emit→render contract test — guards against builder↔renderer schema drift.').

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**Clean, well-scoped crash-fix PR. Two production bugs (wrong label field path, wrong URL construction) are fixed with minimal, targeted changes; a contract test closes the CI gap that let the drift go undetected. No blocking issues found.**

### What this PR is doing
**Intent:** Fix two on-device home-page crashes in `@one-impression/sdui-runtime` by correcting flat `TextSchema` label reads across 7 renderers (`v.label.data.text` → `v.label.text`) and fixing `bff_call` endpoint resolution (logical id → `EndpointPaths` path lookup instead of treating the id as a literal URL path), plus tightening the `sdk-native-sdui` peer-dep to `^2.6.0` and adding an emit→render contract test to catch future drift in CI.
**Delivered:** The diff fully delivers the stated intent. All 7 renderer files are corrected (Tab, Tag, Chip, Checkbox, Radio, SelectableItem, Input). The `bff-call.ts` handler now resolves through `EndpointPaths`, trims trailing slashes, and throws loudly on unregistered ids. The peer-dep is tightened in both `package.json` and `package-lock.json`. The contract test covers schema round-trips for all home node types plus the flat-label regression assertion. Four new `bff_call` URL-resolution tests are added. No acceptance items are missing; no scope overshoot is visible.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- The fix is surgical — exactly the two broken field-access patterns are corrected across all affected renderers, with no unrelated changes bundled in.
- The emit→render contract test is a genuinely useful addition: it round-trips real SDK builders through their own Zod schemas AND asserts the exact field path the renderers read, so future schema drift will fail in CI rather than on a device.
- Tightening the peer-dep from `>=1.0.0` to `^2.6.0` is the right call — it surfaces the version gap at install time and is correctly mirrored as a devDependency so the `EndpointPaths` import resolves at build/test time.
- The `throw new Error(...)` on an unregistered endpoint id is the correct failure mode — loud and immediate rather than silently constructing a wrong URL.
- The changeset entry is thorough and accurately describes all four fixes, making the release notes self-contained.

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

File: packages/sdui-runtime/src/__tests__/emit-render-contract.test.ts (hunk 1 above)
Comment (zenith-oi) [inline packages/sdui-runtime/src/__tests__/emit-render-contract.test.ts]: **FLAG** — [no-task-refs-in-durable-artifacts] The file-level JSDoc opens with `* Emit→render contract test (CR-WF3).` — `CR-WF3` is a task/work-item reference. Per the `no-task-refs-in-durable-artifacts` pattern, task IDs must not appear in source files; they belong in commit messages and PR descriptions. Replace with a timeless description of what the test does, e.g. `* Emit→render contract test — catches builder↔renderer schema drift in CI.`

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**Clean crash-fix PR that closes two on-device regressions and adds structural anti-regression coverage. One carry-over FLAG from the previous round remains unaddressed (task ref `CR-WF3` in the test file header comment). No new issues introduced.**

### What this PR is doing
**Intent:** Align `@one-impression/sdui-runtime` with the `@one-impression/sdk-native-sdui` wire contract by fixing two on-device crashes (flat label reads and bff_call endpoint resolution), tightening the peer-dep range, and adding anti-regression tests — inferred from diff + body.
**Delivered:** The diff fully delivers all four stated fixes: (1) all 7 renderer files updated from `v.label.data.text` → `v.label.text`; (2) `bff-call.ts` now resolves through `EndpointPaths` with trailing-slash trim and loud throw on unknown id; (3) peer-dep tightened to `^2.6.0` and added as devDependency; (4) emit→render contract test and 4 new bff_call URL-resolution tests added. CI workflow updated to authenticate against the GitHub npm registry so the new private devDependency resolves in CI. No acceptance items are missing.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- The `EndpointPaths` resolution approach is exactly right — single source of truth shared with the gateway, loud throw on unregistered id rather than silently constructing a wrong URL.
- Adding `sdk-native-sdui` as a devDependency (not just a peerDependency) so the import resolves at build/test time is a clean pattern that surfaces drift at install rather than on-device.
- The emit→render contract test is well-structured: it separates schema round-trip (builder self-consistency) from field-shape regression (renderer reads the right path), and the inline comment explaining why renderers can't be mounted in this harness is genuinely useful for future contributors.
- Trailing-slash trim on `bffBaseUrl` is handled defensively with a regex replace rather than a conditional, which correctly handles both trailing-slash and no-trailing-slash base URLs without branching.

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

File: packages/sdui-runtime/src/__tests__/emit-render-contract.test.ts (hunk 1 above)
Comment (zenith-oi) [inline packages/sdui-runtime/src/__tests__/emit-render-contract.test.ts]: **FLAG** — [no-task-refs-in-durable-artifacts] The file-level JSDoc opens with `* Emit→render contract test (CR-WF3).` — `CR-WF3` is a task/work-item reference embedded in a durable source file. Per the `no-task-refs-in-durable-artifacts` pattern, task IDs belong in commit messages and PR descriptions, not in code or test files. Replace with a timeless description of what the test does, e.g. `* Emit→render contract test — catches builder↔renderer schema drift in CI.`

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**Solid crash-fix PR that correctly aligns the runtime with the wire contract on two independent failure vectors. The one carry-over issue from the previous review round (task reference `CR-WF3` in a durable source file) remains unaddressed.**

### What this PR is doing
**Intent:** Fix two on-device home-page crashes in `@one-impression/sdui-runtime` by correcting flat `TextSchema` label reads across 7 renderers (`v.label.data.text` → `v.label.text`) and fixing `bff_call` endpoint resolution (logical id → path via `EndpointPaths`), while tightening the `sdk-native-sdui` peer-dep to `^2.6.0` and adding anti-regression contract tests to catch future drift in CI.
**Delivered:** The diff fully delivers the stated intent. All 7 renderer files are corrected (Tab, Tag, Chip, Checkbox, Radio, SelectableItem, Input). The `bff-call.ts` handler now resolves via `EndpointPaths`, trims trailing slashes, and throws on unregistered ids. The peer-dep is tightened in both `package.json` and `package-lock.json`. The CI workflow is updated to authenticate against the GitHub npm registry so the new devDependency resolves. Four contract tests and four bff_call URL-resolution tests are added. No acceptance items are missing.

> ℹ️ 1 previous Zenith review dismissed. Inline comments from those reviews are superseded — only comments from **this review** are current.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- The `EndpointPaths` resolution approach is clean — using the shared map as the single source of truth between gateway and runtime is exactly the right design, and the loud throw on unregistered ids prevents silent 404s.
- Adding the sdk-native-sdui devDependency alongside the tightened peerDependency is the correct two-pronged approach: build/test time resolution + install-time contract enforcement.
- The CI workflow change (registry-url + NODE_AUTH_TOKEN) is minimal and correct — only the necessary lines added to unblock the private package install.
- The contract test structure is well-reasoned: schema round-trips catch builder drift, and the explicit `data.label.data` absence assertion directly encodes the crashing assumption that was wrong, making future regressions immediately visible.

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

File: packages/sdui-runtime/src/action-engine/handlers/bff-call.ts [hunk 2]
```diff
@@ -23,16 +23,29 @@ export async function handleBffCall(
...
+    // Should never happen — endpoint is a typed EndpointId. Fail loudly rather
+    // than silently constructing a wrong URL.
+    throw new Error(`bff_call: no path registered for endpoint id "${payload.endpoint}"`);
+  }
+
+  // Substitute path params into the resolved path (e.g. "/v1/.../{id}").
+  let endpointPath = path;
   if (payload.path_params) {
     for (const [key, value] of Object.entries(payload.path_params)) {
       endpointPath = endpointPath.replace(`{${key}}`, encodeURIComponent(value));
```
Comment (zenith-oi) [inline packages/sdui-runtime/src/action-engine/handlers/bff-call.ts:41]: **FLAG** — [error-handling-completeness] After substituting known path params, any `{placeholder}` that was declared in the template but absent from `payload.path_params` will remain literally in the URL (e.g. `/v1/creator/campaigns/{id}` if `id` is missing). The current code silently sends a request to a URL containing a brace-wrapped token, which the server will 404 or 400 on with no useful client-side error. Consider scanning the resolved path for remaining `{...}` tokens after substitution and throwing a descriptive error — the same "fail loudly" philosophy already applied to unregistered endpoint ids.

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**Solid crash-fix PR that correctly aligns the runtime with the wire contract, adds meaningful regression coverage, and closes the version-gap that allowed the drift to go undetected. No blocking issues found; one minor flag on unsubstituted path-param placeholders in the resolved URL.**

### What this PR is doing
**Intent:** Fix two on-device home-page crashes in `@one-impression/sdui-runtime` by correcting flat `TextSchema` label reads across 7 renderers (`v.label.data.text` → `v.label.text`) and fixing `bff_call` endpoint resolution (logical id → `EndpointPaths` lookup instead of treating the id as a literal path), plus tightening the `sdk-native-sdui` peer-dep to `^2.6.0` and adding an emit→render contract test to catch future drift in CI.
**Delivered:** Fully delivered. All four stated fixes are present in the diff: (1) flat label reads corrected in all 7 renderers (Tab, Tag, Chip, Checkbox, Radio, SelectableItem, Input); (2) `bff_call` now resolves through `EndpointPaths`, trims trailing slash, throws on unknown id; (3) peer-dep tightened and added as devDependency; (4) emit→render contract test added with 4 cases plus 4 new `bff_call` URL-resolution tests. CI workflow updated to authenticate against the GitHub npm registry so the private `sdk-native-sdui` package resolves during `npm ci`. No scope overshoot detected.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- The emit→render contract test is exactly the right architectural move — it round-trips the SDK builder output through the SDK schema AND asserts the exact field shape the renderer reads, so future schema/renderer drift surfaces in CI rather than on-device. The comment block explaining why renderers can't be mounted in the node:test runner is also genuinely useful.
- Throwing loudly on an unregistered endpoint id (`throw new Error(...)`) rather than silently constructing a wrong URL is the correct defensive posture and matches the repo's stated philosophy.
- Tightening the peer-dep from `>=1.0.0` to `^2.6.0` and adding the package as a devDependency so the import resolves at build/test time is the right structural fix — it moves the failure from runtime to install time.

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

File: packages/sdui-runtime/src/__tests__/emit-render-contract.test.ts (hunk 1 above)
Comment (zenith-oi) [inline packages/sdui-runtime/src/__tests__/emit-render-contract.test.ts]: **FLAG** — [no-task-refs-in-durable-artifacts] The file-level JSDoc opens with `* Emit→render contract test (CR-WF3).` — `CR-WF3` is a task/work-item reference embedded in a durable source file. Per the `no-task-refs-in-durable-artifacts` pattern, task/brief/PR identifiers must not appear in code files; they belong in commit messages and PR descriptions only. Replace with a timeless description of what the test does, e.g. `* Emit→render contract test — catches builder↔renderer schema drift in CI.`

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

> ⚠️ **Round 4 — engineer attention requested.** The reviewer-engineer loop has reached round 4; this is the threshold where auto-iteration typically stops being productive (either the review is missing context the engineer has, or there's a genuine design call to make). Auto-fixes are still posted, but please review this thread end-to-end rather than relying on another round to converge.

**Clean bug-fix PR that corrects two crash-class regressions and closes the version-gap that allowed them. Code is well-structured, tests are meaningful, and the changeset is appropriately scoped. One carry-over FLAG from the previous round remains unaddressed (task reference in a durable file header).**

### What this PR is doing
**Intent:** Fix two on-device home-page crashes in `@one-impression/sdui-runtime` by correcting flat `TextSchema` label reads across 7 renderers (`v.label.data.text` → `v.label.text`) and fixing `bff_call` endpoint resolution to use `EndpointPaths` instead of treating the logical id as a literal path, plus tightening the `sdk-native-sdui` peer-dep to `^2.6.0` and adding an emit→render contract test to catch future drift in CI.
**Delivered:** The diff fully delivers the stated intent. All 7 renderer files are corrected (Tab, Tag, Chip, Checkbox, Radio, SelectableItem, Input). The `bff-call.ts` handler now resolves via `EndpointPaths`, substitutes path params, guards unsubstituted params, and trims trailing slashes. The peer-dep and devDependency are both updated. The contract test covers schema round-trips and the flat-label regression. CI is updated to authenticate against the GitHub npm registry so the new devDependency resolves. No acceptance items are missing.

> ℹ️ 1 previous Zenith review dismissed. Inline comments from those reviews are superseded — only comments from **this review** are current.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- The unsubstituted path-param guard (`/\{[^}]+\}/` check after substitution) is a good defensive addition — it surfaces misconfigured actions loudly rather than sending a literal `{id}` to the server.
- Adding `sdk-native-sdui` as both a devDependency (for build/test resolution) and tightening the peerDependency range to `^2.6.0` is exactly the right two-pronged fix — it surfaces schema drift at install time for consumers and at build time for this package.
- The emit→render contract test is well-scoped: it explicitly documents why full renderer mounting isn't possible in the current test harness and focuses on the specific field-shape invariant that caused the crash, making it a durable regression guard rather than a one-off smoke test.

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

File: packages/sdui-runtime/src/action-engine/handlers/bff-call.ts [hunk 3]
```diff
@@ -23,16 +23,40 @@ export async function handleBffCall(
 ): Promise<void> {
   const payload = BffCallPayloadSchema.parse(action.payload);
 
-  // Build the URL, substituting path params.
-  let endpointPath = payload.endpoint as string;
+  // Resolve the logical endpoint id (e.g. "creator.home.tab") to its request
+  // path (e.g. "/v1/creator/home/tab") via the EndpointPaths map — the single
+  // source of truth shared with the gateway. The id is NOT a path; treating it
+  // as one would request "/creator.home.tab" and 404.
+  const path = EndpointPaths[payload.endpoint];
```
Comment (zenith-oi) [inline packages/sdui-runtime/src/action-engine/handlers/bff-call.ts:30]: **FLAG** — [structured-error-types] `EndpointPaths[payload.endpoint]` returns `string | undefined` (or the equivalent from the SDK's map type). The guard on line 31 (`if (!path)`) correctly catches the undefined case and throws a plain `new Error(...)`. For a runtime library this is acceptable, but the thrown error is a generic `Error` with no structured kind discriminator — if the action engine has a top-level error handler that maps error types to UI states (e.g. toast vs hard crash), it cannot distinguish an unregistered-endpoint programming error from a network failure. Consider throwing a typed error (e.g. `class BffCallConfigError extends Error { kind = 'unregistered_endpoint' as const }`) so the engine can route it correctly. Not blocking — the current throw is strictly better than the silent 404 it replaces.

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**Solid crash-fix PR that aligns the runtime with the wire contract, adds meaningful regression coverage, and closes the version-gap that let the drift accumulate. No blocking issues found; one flag on missing error handling for the `EndpointPaths` lookup result type.**

### What this PR is doing
**Intent:** Fix two on-device home-page crashes in `@one-impression/sdui-runtime` by (1) correcting flat `TextSchema` label reads across 7 renderers (`v.label.data.text` → `v.label.text`), (2) resolving logical endpoint IDs through `EndpointPaths` in `bff_call` instead of treating them as literal URL paths, (3) tightening the `sdk-native-sdui` peer-dep to `^2.6.0`, and (4) adding an emit→render contract test to catch future drift in CI.
**Delivered:** All four stated fixes are present and correct. FIX 1: all 7 renderer files updated (Tab, Tag, Chip, Checkbox, Radio, SelectableItem label+subtitle, Input). FIX 2: `bff-call.ts` now resolves via `EndpointPaths`, substitutes path params, guards unsubstituted params, and trims trailing slash. FIX 3: `package.json` peer-dep and dev-dep both updated; lock-file reflects the new `sdk-native-sdui@2.6.0` dev install. FIX 4: `emit-render-contract.test.ts` covers schema round-trips and flat-label regression; 4 new `bff_call` URL-resolution tests added. CI workflow updated to authenticate against the GitHub npm registry so the private `@one-impression/sdk-native-sdui` package resolves in CI. No linked task page was provided, so acceptance items cannot be cited by number; intent is fully delivered per the PR body.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- The `no-task-refs-in-durable-artifacts` pattern is respected: the changeset and test file comments describe what the code does (flat TextSchema, EndpointPaths resolution) rather than citing task/brief IDs.
- The unsubstituted-path-param guard (`/\{[^}]+\}/` check after substitution) is a good defensive addition — it surfaces misconfigured actions loudly at call time rather than sending a literal `{id}` to the server.
- Adding `sdk-native-sdui` as a devDependency alongside the tightened peerDependency is the correct dual-registration pattern: the peer constraint enforces the consumer's install, while the dev entry makes the import resolve at build/test time without bundling the package.
- The CI workflow change (`registry-url` + `NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}`) is the minimal correct fix to allow `npm ci` to pull the private GitHub Packages dependency in CI — no secrets are hardcoded.
- The emit→render contract test is well-scoped and clearly documented: it explicitly calls out that it cannot mount renderers (RN transform limitation) and what the next expansion step should be, making the test's intentional boundaries visible to future contributors.

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

File: packages/sdui-runtime/src/__tests__/emit-render-contract.test.ts [hunk 4]
```diff
@@ -0,0 +1,155 @@
+/**
+ * Emit→render contract test.
```
Comment (zenith-oi) [inline packages/sdui-runtime/src/__tests__/emit-render-contract.test.ts:2]: **FLAG** — [no-task-refs-in-durable-artifacts] The file-level JSDoc block opens with `* Emit→render contract test (CR-WF3).` — `CR-WF3` is a task/work-item reference embedded in a durable source file. Per the pattern, task/brief/work-item IDs belong in commit messages and PR descriptions, not in source files that outlive the PR. The fix is mechanical: drop `(CR-WF3)` from the comment and describe what the test does in domain terms (which the rest of the comment already does well).

**Review by zenith-oi — APPROVED** [review]
## Zenith Code Review

> ⚠️ **Round 5 — engineer attention requested.** The reviewer-engineer loop has reached round 5; this is the threshold where auto-iteration typically stops being productive (either the review is missing context the engineer has, or there's a genuine design call to make). Auto-fixes are still posted, but please review this thread end-to-end rather than relying on another round to converge.

**Clean, well-scoped bug-fix PR that closes two on-device crashes and adds meaningful anti-regression coverage. One pre-existing FLAG from the previous review round (task reference in a durable artifact) remains unaddressed.**

### What this PR is doing
**Intent:** Align `@one-impression/sdui-runtime` with the `@one-impression/sdk-native-sdui` wire contract by fixing two on-device crashes (flat label reads and bff_call endpoint resolution), tightening the peer-dep range, and adding anti-regression contract tests.
**Delivered:** The diff fully delivers the stated intent. FIX 1 (flat label reads) is applied across all 7 renderers. FIX 2 (bff_call endpoint resolution via EndpointPaths) is implemented with path-param substitution, unsubstituted-param guard, trailing-slash trim, and unregistered-id guard. FIX 3 (peer-dep tightened to ^2.6.0 + devDependency added) is present in both package.json and package-lock.json. FIX 4 (anti-regression tests) ships 4 contract tests and 5 bff_call URL-resolution tests. CI is updated to authenticate against the private GitHub npm registry so the new devDependency resolves. No acceptance items are deferred or missing.

> ℹ️ 1 previous Zenith review dismissed. Inline comments from those reviews are superseded — only comments from **this review** are current.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- The EndpointPaths resolution in bff-call.ts is exactly right: fail-loud on unregistered ids, fail-loud on unsubstituted path params, trailing-slash trim — all three edge cases handled and each covered by a dedicated test.
- The emit→render contract test is a genuinely useful addition: it round-trips builders through their own schemas AND asserts the exact field shape the renderers read, catching builder↔renderer drift in CI rather than on-device.
- Tightening the peerDependency from `>=1.0.0` to `^2.6.0` and adding the package as a devDependency is the right structural fix — it surfaces schema/renderer drift at install time instead of at runtime.

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

**Review by zenith-oi — APPROVED** [review]
## Zenith Code Review

**Targeted crash-fix PR that aligns the SDUI runtime with the `sdk-native-sdui` wire contract. All four stated fixes are present and the new tests directly lock the regressions. One real issue worth flagging: the `encodeURIComponent` path-param substitution happens before the unsubstituted-param guard, so a param value that happens to contain `{` or `}` characters could theoretically confuse the guard — but more practically, the guard regex `\{[^}]+\}` runs on the already-substituted string, which is the correct and intended behaviour, so this is actually fine. The one genuine concern is that `EndpointPaths` is imported as a value from the peer dependency at module load time; if a consumer installs a `sdk-native-sdui` version that doesn't export `EndpointPaths`, the import will fail at runtime rather than at install time — but the peer-dep tightening to `^2.6.0` is precisely the mitigation for this, and the devDependency addition ensures CI catches it. No blocking issues found.**

### What this PR is doing
**Intent:** Fix two on-device home-page crashes in `@one-impression/sdui-runtime` by (1) correcting flat `TextSchema` label reads across 7 renderers, (2) resolving logical endpoint IDs through `EndpointPaths` in `bff_call`, (3) tightening the `sdk-native-sdui` peer-dep to `^2.6.0`, and (4) adding an emit→render contract test suite to catch future drift in CI.
**Delivered:** The diff fully delivers all four stated fixes. FIX 1: all 7 renderer files (`Tab`, `Tag`, `Chip`, `Checkbox`, `Radio`, `SelectableItem`, `Input`) are updated from `v.label.data.text` → `v.label.text` (and `v.subtitle.data.text` → `v.subtitle.text`). FIX 2: `bff-call.ts` now resolves via `EndpointPaths`, substitutes path params, guards against unsubstituted params, and trims trailing slash. FIX 3: `package.json` peer-dep raised to `^2.6.0` and added as devDependency. FIX 4: `emit-render-contract.test.ts` (4 test cases) and 5 new `bff-call` tests added. CI workflow updated to authenticate against the GitHub npm registry so the private `sdk-native-sdui` package resolves. No linked task page or brief was provided, so acceptance items are inferred from the PR body — all stated items are present in the diff.

No issues found.

**Positive:**
- The emit→render contract test is an excellent architectural decision — it round-trips builder output through the schema AND asserts the exact field path the renderer reads, so future schema drift surfaces in CI rather than on-device. The `assert('data' in label, false)` negative assertion is particularly sharp.
- The unsubstituted path-param guard (`/\{[^}]+\}/` after substitution) is a good defensive addition — it turns a silent 404/400 from the server into a loud client-side throw with a clear message naming the offending param.
- Adding `sdk-native-sdui` as both a tightened peerDependency (`^2.6.0`) AND a devDependency at the same version is the correct dual-registration pattern: consumers get an install-time warning if they're on the wrong version, and the build/test environment resolves the import without relying on the peer being present.

**Read against:**
- `PR body — FIX 1 (flat label reads), FIX 2 (bff_call endpoint resolution), FIX 3 (peer-dep lock), FIX 4 (anti-regression test)`
- `packages/sdui-runtime/src/action-engine/handlers/bff-call.ts — EndpointPaths resolution logic`
- `packages/sdui-runtime/src/__tests__/emit-render-contract.test.ts — contract test coverage`

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*
